### PR TITLE
Build NLE (mostly) out-of-dir

### PR DESCRIPTION
**Pull request by heiner**

### CODE REVIEW

**Review by edran — APPROVED** [review]
LGTM -- very clever fix, well done!

We shall make a couple of sacrifices to the CMake gods every month from now on 📿 .

File: dat/CMakeLists.txt [hunk 1]
```diff
@@ -1,15 +1,3 @@
...
-    bogusmon
-    data
-    engrave
-    epitaph
-    oracles
-    options
-    quest.dat
-    rumors)
-set(VARDAT ${VARDATD} ${VARDATND})
-
```
[the remark this answers is not in the document]
Comment (heiner) [reply dat/CMakeLists.txt:12]: Yes I saw that and it makes a lot of sense too. But in the end we also need so many new variables that it became confusing and not all of them were used in the end.

**Comment by edran** [comment]
Sigh: https://app.circleci.com/pipelines/github/facebookresearch/nle/414/workflows/9c77fa62-4a1d-4252-9cc5-a519be50ac2e/jobs/1684

**Comment by heiner** [comment]
> Sigh: https://app.circleci.com/pipelines/github/facebookresearch/nle/414/workflows/9c77fa62-4a1d-4252-9cc5-a519be50ac2e/jobs/1684

No idea what this is but it's not related to this change. Perhaps the test I wrote there isn't quite correct. Will re-run which should make it go away.

**Pull request merged**